drivers/hmis/app/models/hmis/workflow_execution/engine.rb

```diff
@@ -40,6 +40,8 @@ def enable_step!(node)
         raise "Failed to reopen step #{step.id} because it had an irreversible side effect. This indicates a misconfigured workflow."
       end
 
+      return step if step.open? # If the step is already available, no action needed
```

Comment:
I guess we don't need to also check that `step.persisted?` because if it was new, `find_or_initialize_by` would initialize it with `unavailable` state? I think thats right, but just calling it out.